packages/connect/src/implementation.ts

```diff
@@ -251,14 +252,36 @@ export type ServiceImplSpec = {
  * Create an MethodImplSpec - a user-provided implementation for a method,
  * wrapped in a discriminated union type along with service and method metadata.
  */
+export function createMethodImplSpec<M extends MethodType>(
+  method: M,
+  impl: MethodImpl<M>,
+): MethodImplSpec;
```

Comment:
I would prefer if we only support `MethodType` in `ConnectRouter.rpc` in this PR to keep the change well-scoped.